Scripts/freight_lem.py

```diff
@@ -0,0 +1,194 @@
+from assignment.emme_assignment import EmmeAssignmentModel
+from assignment.emme_bindings.emme_project import EmmeProject
+from datatypes.purpose import FreightPurpose
+from datahandling.matrixdata import MatrixData
+
+from datahandling.traversaldata import transform_traversal_data
+from utils.freight_costs import calc_rail_cost, calc_road_cost, calc_ship_cost
+from parameters.commodity import commodity_conversion
+
+BASE_FOLDER = Path(__file__).parent
+BASE_ZONEDATA_FILE = "freight_zonedata.gpkg"
+
+
+def main(args):
+    log.info("Starting simulation.")
+    base_zonedata_path = Path(BASE_FOLDER, args.baseline_data_path, BASE_ZONEDATA_FILE)
+    cost_data_path = Path(BASE_FOLDER, args.cost_data_path)
+    results_path = Path(BASE_FOLDER, args.results_path)
+    emme_project_path = Path(BASE_FOLDER, args.emme_path)
+    parameters_path = BASE_FOLDER / "parameters" / "freight"
+    
+    ass_model = EmmeAssignmentModel(EmmeProject(emme_project_path),
+                                    first_scenario_id=args.first_scenario_id,
+                                    save_matrices=args.save_emme_matrices,
+                                    first_matrix_id=args.first_matrix_id)
+    zone_numbers = ass_model.zone_numbers
+    zonedata = FreightZoneData(base_zonedata_path, zone_numbers, "koko_suomi")
+    resultdata = ResultsData(results_path)
+    resultmatrices = MatrixData(results_path / "Matrices" / "koko_suomi")
+    costdata = json.loads(cost_data_path.read_text("utf-8"))
+    purposes = {}
+    for file in parameters_path.rglob("*.json"):
+        commodity_params = json.loads(file.read_text("utf-8"))
+        commodity = commodity_params["name"]
+        if commodity not in ("marita", "kalevi"):
+            continue
+        purposes[commodity] = FreightPurpose(commodity_params, zonedata, resultdata)
+    purps_to_assign = list(filter(lambda purposes: purposes[0] in
+                                  list(purposes), args.specify_commodity_names))
+    ass_model.prepare_freight_network(costdata["car_cost"], purps_to_assign)
+    impedance = ass_model.freight_network.assign()
+    zeros = numpy.zeros([len(zone_numbers), len(zone_numbers)])
+    impedance = {
+        "truck": {
+            "dist": impedance["dist"]["truck"],
+            "time": impedance["time"]["truck"],
+            "toll": zeros
+        },
+        "freight_train": {
+            "dist": impedance["dist"]["freight_train"],
+            "time": impedance["time"]["freight_train"],
+        },
+        "ship": {
+            "dist": impedance["dist"]["ship"],
+            "channel": zeros
+        },
+        "freight_train_aux": {
+            "dist": impedance["aux_dist"]["freight_train"],
+            "time": impedance["aux_time"]["freight_train"],
+            "toll": zeros
+        },
+        "ship_aux": {
+            "dist": impedance["aux_dist"]["ship"],
+            "time": impedance["aux_time"]["ship"],
+            "toll": zeros
+        }
```

Comment:
I suggest integrating `impedance_aux` into `impedance` in `calc_rail_cost`, `calc_ship_cost` and `get_aux_cost`, so that names of "aux_dist" and "aux_time" would not change. The name changes "aux_dist" -> "dist" and "aux_time" -> "time" could happen inside `get_aux_cost`. In that way you would not have to do this hard-coded remapping here. This would then be enough here:

```
impedance = {mode: {mtx_type: impedance[mtx_type][mode] for mtx_type in impedance
    if mode in impedance[mtx_type]} for mode in ("truck", "freight_train", "ship")}
```